src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -286,7 +287,43 @@ app.directive('dashboardApp', function ($injector) {
         }).catch(notify.fatal);
       };
 
+      $scope.showFilterBar = () => filterBar.getFilters().length > 0 || !$scope.fullScreenMode;
+      let onRouteChange;
+      const setFullScreenMode = (fullScreenMode) => {
+        $scope.fullScreenMode = fullScreenMode;
+        dashboardState.setFullScreenMode(fullScreenMode);
+        chrome.setVisible(!fullScreenMode);
+        $scope.$root.$broadcast('globalNav:update');
```

Comment:
Using the native `resize` event feels equally gross because it's normally fired by the browser itself, and it feels like we'd be hijacking this as well.

Since the `dashboard-grid` is a directive within the `dashboard` we shouldn't have to resort to using a `$scope.$root.$broadcast` and should be able to just call a function that we pass to it if we need to trigger the resize.